custom/icds_reports/static/js/angular-services/baseControllers.service.js

```diff
@@ -1,10 +1,26 @@
 /* global d3, moment */
 
 window.angular.module('icdsApp').factory('baseControllersService', function() {
+    var BaseFilterController = function ($scope, $routeParams, $location, dateHelperService, storageService) {
```

Comment:
This is another new paradigm of porting a subset of functionality from `BaseController` that was copy/pasted in other places into a separate class so it could be reused by both `BaseController` and some of those other components (see next commit for the payoff).